### PR TITLE
Added data source google kms secret asymmetric

### PR DESCRIPTION
**Pull request by mvanholsteijn**
This PR add the data source `google_kms_secret_asymmetric`. I chose to put `_asymmetric` as a suffix to ensure that in the documentation it will appear next to `google_kms_secret`.

This partly deals with issue https://github.com/hashicorp/terraform-provider-google/issues/6462

### CODE REVIEW

**Review by dennisvink — APPROVED** [review]

**Comment by mvanholsteijn** [comment]
Hi @melinathI am looking forward to your review!  Just let me know if you have any questions for me, or whether I have to change things here.

File: website/docs/d/kms_secret_asymmetric.html.markdown [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+This will result in a Cloud SQL user being created with password `my-secret-password`.
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `ciphertext` (Required) - The ciphertext to be decrypted, encoded in base64
+* `crypto_key_version` (Required) - The id of the CryptoKey version that will be used to
+  decrypt the provided ciphertext. This is represented by the format
+  `projects/{project}/locations/{location}/keyRings/{keyring}}/cryptoKeys/{key}}/cryptoKeyVersions/{version}`.
```
Comment (melinath) [inline website/docs/d/kms_secret_asymmetric.html.markdown:141]: typo: a couple extra closing curly braces

**Review by melinath — CHANGES_REQUESTED** [review]
Hello, thanks for your PR! This API looks like a GA api; I'll need to upstream your changes to https://github.com/GoogleCloudPlatform/magic-modules, after which they'll be put into both the beta and GA providers. Does that seem correct?

This implementation seems fine; I just have a couple questions.

File: google-beta/config.go [hunk 2]
```diff
@@ -447,6 +450,26 @@ func (c *Config) NewKmsClient(userAgent string) *cloudkms.Service {
...
+		log.Printf("[WARN] Error creating client kms invalid base path url %s, %s", c.KMSBasePath, err)
+		return nil
+	}
+	endpoint := u.Host
+	if u.Port() == "" {
+		endpoint = fmt.Sprintf("%s:443", u.Host)
+	}
+
+	log.Printf("[INFO] Instantiating Google Cloud KMS client for path on endpoint %s", endpoint)
+	clientKms, err := kms.NewKeyManagementClient(ctx, option.WithUserAgent(userAgent), option.WithEndpoint(endpoint))
```
[the remark this answers is not in the document]
Comment (mvanholsteijn) [reply google-beta/config.go:465]: I am using the library specified by the [documentation to use](https://cloud.google.com/kms/docs/reference/libraries#client-libraries-install-go). The package documentation of [google.golang.org/api/cloudkms/v1 ](https://pkg.go.dev/google.golang.org/api@v0.42.0/cloudkms/v1#section-documentation) also states that the  package is deprecated.

File: google-beta/data_source_google_kms_secret_asymmetric.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	d.SetId(fmt.Sprintf("%s:%x:%s", cryptoKeyVersion, ciphertextCRC32C, base64CipherText))
+	return nil
+}
+
+func removeWhiteSpaceFromString(s string) string {
+	whitespaceRegexp := regexp.MustCompile(`(?m)[\s]+`)
+	return whitespaceRegexp.ReplaceAllString(s, "")
+}
+
+func validateBase64WithWhitespaces(i interface{}, val string) ([]string, []error) {
```
[the remark this answers is not in the document]
Comment (mvanholsteijn) [reply google-beta/data_source_google_kms_secret_asymmetric.go:127]: The validateBase64String does not allow for whitespace in the base64 encoding. As the cyphertext is quite long, I wanted to be able to specify the output of openssl base64 which does include whitespace.

File: website/docs/d/kms_secret_asymmetric.html.markdown (hunk 1 above)
Comment (mvanholsteijn) [reply website/docs/d/kms_secret_asymmetric.html.markdown]: fixed

**Review by melinath — APPROVED** [review]
LGTM - I'll get to work upstreaming.

**Comment by mvanholsteijn** [comment]
@melinath great! Thank you!

**Pull request merged**